graphics/fonts/bdf.cpp

```diff
@@ -787,57 +791,70 @@ BdfFont *BdfFont::scaleFont(const BdfFont *src, int newSize) {
 		const BdfBoundingBox &box = data.boxes ? data.boxes[i] : data.defaultBox;
 		const BdfBoundingBox &srcBox = data.boxes ? src->_data.boxes[i] : src->_data.defaultBox;
 
+		int grayLevel = box.height * box.width / 3;
+		int dstPitch = (box.width + 7) / 8 ;
+
+#if DRAWDEBUG
+		int ccc = 'a';
+#endif
 		if (src->_data.bitmaps[i]) {
-			const int bytes = ((box.width + 7) / 8) * box.height; // Dimensions have been already corrected
+			int *dstGray;
+			Graphics::Surface *srcSurf = srcCopy->scaleSingleGlyph(box.width, box.height, grayLevel, i + src->_data.firstCharacter, 
+																   box.xOffset, src->_data.height, srcBox.width, scale, dstGray);
+			const int bytes = dstPitch * box.height;
 			bitmaps[i] = new byte[bytes];
-
-			int srcPitch = (srcBox.width + 7) / 8;
-			int dstPitch = (box.width + 7) / 8;
-
 			byte *ptr = bitmaps[i];
-
 			for (int y = 0; y < box.height; y++) {
-				const byte *srcd = (const byte *)&src->_data.bitmaps[i][((int)((float)y / scale)) * srcPitch];
+				byte *srcd = (byte *)srcSurf->getBasePtr(0, y);
 				byte *dst = ptr;
 				byte b = 0;
-
-				for (int x = 0; x < box.width; x++) {
+				for (int x = 0; x < box.width; x++,srcd++) {
 					b <<= 1;
-
-					int sx = (int)(roundf((float)x / scale));
-
-					if (srcd[sx / 8] & (0x80 >> (sx % 8))) {
+					if (*srcd == 1) {
 						b |= 1;
-#if DRAWDEBUG
-						debugN("#");
-					} else {
-						debugN(" ");
-#endif
 					}
-
 					if (x % 8 == 7) {
 						*dst++ = b;
 						b = 0;
 					}
-				}
-#if DRAWDEBUG
-				debug("");
-#endif
-
+				}	
 				if (((box.width - 1) % 8)) {
 					b <<= 7 - ((box.width - 1) % 8);
 					*dst = b;
 				}
-
 				ptr += dstPitch;
+#if DRAWDEBUG
+				if (i == ccc) {
+					debugN("--> %d ", grayLevel);
+					grayPtr = &dstGray[y * box.width];
+					for (int x = 0; x < ; box.widthx++, grayPtr++)
+						debugN("%c", *grayPtr > grayLevel ? '@' : '_');
+					debugN("\n");
+					debugN("***");
+				}
+#endif
 			}
-
+#if DRAWDEBUG
+			if (i == ccc) {
+				warning("BdfFont::scaleFont(): %d %d", data.height,box.height);
```

Comment:
Do you need this as a warning? And if you do, the code formatting is off.

Reply:
yes that is not required, removed it now